DOC-REFACTOR: 1/n Refactoring the documentation - added first stab fo… (#1761)
modin-project/modin: docs/developer/contributing.rst

.. code-block:: bash

   git config --global user.name "Awesome Developer"
   git config --global user.email "awesome.developer.@example.org"

If you have authored a commit that is missing the signed-off-by line, you can amend your
commits and push them to GitHub.

.. code-block:: bash

   git commit --amend --signoff

If you've pushed your changes to GitHub already you'll need to force push your branch
after this with ``git push -f``.

Development Dependencies
------------------------

We recommend doing development in a virtualenv or conda environment, though this decision 
is ultimately yours. You will want to run the following in order to install all of the required
dependencies for running the tests and formatting the code:

.. code-block:: bash

  pip install -r requirements.txt

For developments under Windows, dependencies can be found in 'env_windows.yml' file. 

Code Formatting and Lint
------------------------

We use black_ for code formatting. Before you submit a pull request, please make sure
that you run the following from the project root:

.. code-block:: bash

  black modin/

We also use flake8_ to check linting errors. Running the following from the project root

> will ensure that it passes the lint checks on Travis:

Reviewer: ```suggestion
will ensure that it passes the lint checks in the CI testing:
```

.. code-block:: bash

  flake8 .

> We test that this has been run on our `Travis CI`_ test suite. If you do this and find

Reviewer: ```suggestion
We test that this has been run on our GitHub Actions test suite. If you do this and find
```

only modified a small amount of code, it may be sufficient to run a single test or some
subset of the test suite. In order to run a specific test run:

.. code-block:: bash

  pytest modin/pandas/test::test_new_functionality

The entire test suite is automatically run for each pull request.

Contributing a new execution framework or in-memory format
----------------------------------------------------------

If you are interested in contributing support for a new execution framework or in-memory
format, please make sure you understand the architecture_ of Modin.

The best place to start the discussion for adding a new execution framework or in-memory
format is the `developer mailing list`_.

More docs on this coming soon...

.. _Good first issue: https://github.com/modin-project/modin/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue+%3Abeginner%3A%22
.. _Documentation: https://github.com/modin-project/modin/issues?q=is%3Aissue+is%3Aopen+label%3A%22documentation+%3Abookmark_tabs%3A%22
.. _architecture: architecture.html
.. _internal methods:
.. _black: https://github.com/ambv/black
.. _flake8: http://flake8.pycqa.org/en/latest/

> .. _Travis CI: https://travis-ci.org/

Reviewer: ```suggestion
```
